modules/nf-core/cnvkit/batch/main.nf

```diff
@@ -63,133 +64,39 @@ process CNVKIT_BATCH {
-            $tumor_out \\
-            $normal_args \\
-            $fasta_args \\
-            $reference_args \\
-            $target_args \\
-            --processes $task.cpus \\
-            $args
-
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
-            cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
-        END_VERSIONS
-        """
-    }
-    // tumor_only bam_input
-    else if(tumor_bam && !normal_exists){
-        """
-        cnvkit.py \\
-            batch \\
-            $tumor_out \\
-            $normal_args \\
-            $fasta_args \\
-            $reference_args \\
-            $target_args \\
-            --processes $task.cpus \\
-            $args
-
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
-        END_VERSIONS
-        """
-    }
-    // germline mode cram_input
-    // normal_args must be --normal []
-    else if (normal_cram && !tumor_exists){
-        """
-        samtools view -T $fasta $fai_reference $normal -@ $task.cpus -o $tumor_out
-
-        cnvkit.py \\
-            batch \\
-            $tumor_out \\
-            $normal_args \\
-            $fasta_args \\
-            $reference_args \\
-            $target_args \\
-            --processes $task.cpus \\
-            $args
-
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
-            cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
-        END_VERSIONS
-        """
-    }
-    // germline mode bam_input
-    else if (normal_bam && !tumor_exists){
-        """
-        cnvkit.py \\
-            batch \\
-            $tumor_out \\
-            $normal_args \\
-            $fasta_args \\
-            $reference_args \\
-            $target_args \\
-            --processes $task.cpus \\
-            $args
-
-        cat <<-END_VERSIONS > versions.yml
-        "${task.process}":
-            cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
-        END_VERSIONS
-        """
+    if (generate_pon && !tumor_exists && normal_bam){
+        def pon_input = normal.collect("$it").join(' ')
+        normal_args = "--normal $pon_input"
     }
 
+    def target_args = targets ? "--targets $targets" : ""
+    def reference_args = reference ? "--reference $reference" : ""
+
+    def samtools_cram_convert = ''
+    samtools_cram_convert += normal_cram ? "samtools view -T $fasta $fai_reference $normal -@ $task.cpus -o $normal_out\n" : ''
+    samtools_cram_convert += tumor_cram ? "samtools view -T $fasta $fai_reference $tumor -@ $task.cpus -o $tumor_out\n" : ''
+    def samtools_version = normal_cram || tumor_cram ? "samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')" : ''
```

Comment:
I am suspecting that this generates an empty line in the `versions.yml` whenever `samtools` is not used. You could adapt for that adding a line break and a tab and putting everything in one line. Maybe not the most elegant...  
```suggestion
    def samtools_version = normal_cram || tumor_cram ? "samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$/\n\t/')" : ''
```